cli/aps_utils.py

```diff
@@ -1,36 +1,57 @@
+# Copyright (c) 2019 - 2021. Verimatrix. All Rights Reserved.
+# All information in this file is Verimatrix Confidential and Proprietary.
+
 '''Helper utilities'''
 import base64
 import plistlib
 import logging
 import os
 import shutil
-import sys
-import requests
 
 from zipfile import is_zipfile, ZipFile
 
 from pyaxmlparser import APK
 
+from aps_exceptions import ApsException
+
 LOGGER = logging.getLogger(__name__)
 
-def get_config():
+def disable_boto_logging():
+    '''Disable logging from third party modules'''
+    modules_to_disable = ['boto', 's3transfer', 'urllib3', 'boto3', 'botocore']
+    for name in logging.Logger.manager.loggerDict:
+        if name in modules_to_disable:
+            logging.getLogger(name).setLevel(logging.CRITICAL)
+
+def get_config(args):
 
     config = {}
     config['aws_region'] = 'eu-west-1'
     config['api_gateway_url'] = 'https://aps-api.appshield.verimatrixcloud.net'
     config['access-token-url'] = 'https://api.appshield.verimatrixcloud.net/token'
-#
+
+    if(args.api_gateway_url):
+        config['api_gateway_url'] = args.api_gateway_url
+
+    if(args.access_token_url):
+        config['access-token-url'] = args.access_token_url
+
+
     LOGGER.debug('Constructed config object %s', repr(config))
     return config
 
+def get_api_gw_url(config, rest_api_id):
+    ''''Returns the API Gateway URL to use for a specific path'''
+    return config['api_gateway_url'].format(rest_api_id=rest_api_id)
```

Comment:
I dont get the `.format(rest_api_id=rest_api_id)` syntax... you're doing an assignmentof a variable to itself within a `format` statement?

Reply:
Copy/pasted directly from APS. Maybe @sbergman-vmx  could explain the above ?